app/services/user_merger.rb

```diff
@@ -172,6 +173,29 @@ def merge_user_visits
     DB.exec(sql, source_user_id: @source_user.id, target_user_id: @target_user.id)
   end
 
+  def merge_bookmarks
```

Comment:
Did you try to use the `update_user_id` method? I think it's not necessary to create a `merge_bookmarks` method, send a special message over `MessageBus` and write custom SQL just for bookmarks.

Adding the following to the `update_user_ids` should be enough:

```ruby
    update_user_id(
      :bookmarks,
      conditions: [
        "x.bookmarkable_type = y.bookmarkable_type",
        "x.bookmarkable_id = y.bookmarkable_id"
      ]
    )
```

It takes care of everything.